src/app/app.component.ts

```diff
@@ -80,6 +84,23 @@ export class AppComponent implements OnInit {
       });
   }
 
+  toggle(value: string) {
+    if (value === 'toggle' && !this.window.isXS) {
+      this.sideMenu = !this.sideMenu;
+      this.unPin = this.sideMenu ? 'Hide Menu' : 'Show Menu';
```

Comment:
Why is this called unpin? Surely we can make a dynamic tooltip and get rid of this property?